packages/smart/contracts/turbo/AMMFactory.sol

```diff
@@ -133,7 +133,10 @@ contract AMMFactory is BNum {
         _collateral.transferFrom(msg.sender, address(this), _collateralIn);
         _collateral.approve(address(_marketFactory), MAX_UINT);
         uint256 _sets = _marketFactory.calcShares(_collateralIn);
-        _marketFactory.mintShares(_marketId, _sets, address(this));
+        uint256 _tokensIn = (_sets / _marketFactory.shareFactor()) * _marketFactory.shareFactor();
```

Comment:
You're just making sure sets will by a multiple of shareFactor right? I thought we were going to always send the right amount. I'm fine with us doing it this way though, since the downside is you spend a little less collateral than you intended and get a little less shares.

I would just overwrite _sets though. Unless it's still being used?

Reply:
Ha. I broke _sets out for debugging purposes. Should all be on one line. Good catch!